components/authz-service/server/v2/authz.go

```diff
@@ -59,8 +59,8 @@ func (s *authzServer) ProjectsAuthorized(
 	req *api.ProjectsAuthorizedReq) (*api.ProjectsAuthorizedResp, error) {
 
```

Comment:
The first thing is this function is this:
```
projectsAuthorized, err := s.engine.V2ProjectsAuthorized(ctx,
		engine.Subjects(req.Subjects),
		engine.Action(req.Action),
		engine.Resource(req.Resource),
		engine.ProjectList(req.ProjectsFilter...))
```
But remember the above is being used for both v2.0 and v2.1. So you need to check the version before this call, so that you pass `AllProjectsID` instead of `req.ProjectsFilter` as the last param above. That way, if the system is on v2.0 and the user does include project headers, it will ignore them and ask for all projects in the above call.